roofit/roofitcore/inc/RooFuncWrapper.h

```diff
@@ -64,20 +103,34 @@ class RooFuncWrapper final : public RooAbsReal {
 
    double defaultErrorLevel() const override { return 0.5; }
 
+   void getGradient(double *out) const
+   {
+      std::vector<double> paramVals;
```

Comment:
This `vector` means there is a memory allocation on the heap for every `getGradient` call. Can you maybe move this to be a `mutable std::vector<double> _gradientBuffer;` member?